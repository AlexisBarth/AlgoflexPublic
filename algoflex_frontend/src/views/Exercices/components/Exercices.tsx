import * as React from 'react';
import Box from '@mui/material/Box';
import Collapse from '@mui/material/Collapse';
import IconButton from '@mui/material/IconButton';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import Typography from '@mui/material/Typography';
import Paper from '@mui/material/Paper';
import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
import { useHistory, useParams } from 'react-router-dom';
import { Button, Grid } from '@mui/material';
import { useEffect } from "react";
import { client } from '@services/Axios.client';
import { CodingQuestionInterface } from '@components/interfaces';

function Row(props: { row: CodingQuestionInterface }) {
  const { row } = props;
  const [open, setOpen] = React.useState(false);
  const history = useHistory();

  const faireRedirection = () => {
      let url = "/exercice/"+row.uid;
      history.push(url);
  }

  return (
    <React.Fragment>
      <TableRow hover sx={{ '& > *': { borderBottom: 0, paddingTop: 0 } }} style={{backgroundColor: row.backgroundColor}} onClick={() => setOpen(!open)}>
        <TableCell>
          <IconButton
            aria-label="expand row"
            size="small"
            onClick={() => setOpen(!open)}
          >
            {open ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
          </IconButton>
        </TableCell>
        <TableCell scope="row" align="center">
          <Typography variant="h6" component="div">
            {row.name}
          </Typography>
        </TableCell>
        <TableCell align="right" onClick={faireRedirection}>
          <Button size="small" color="primary" variant="contained" onClick={faireRedirection}>
            GO
          </Button>
        </TableCell>
      </TableRow>
      <TableRow>
        <TableCell style={{ paddingBottom: 0, paddingTop: 0 }} colSpan={6}>
          <Collapse in={open} timeout="auto" unmountOnExit>
            <Box sx={{ margin: 1 }}>
              <Typography variant="h6" gutterBottom component="div">
                {row.name}
              </Typography>
              <Typography gutterBottom component='div'>
                {row.description}
              </Typography>
            </Box>
          </Collapse>
        </TableCell>
      </TableRow>
    </React.Fragment>
  );
}

export default function Exercices() {
  let idPage = useParams<any>();
  const [queryData, setQueryData] = React.useState<CodingQuestionInterface[]>([]);
  const [theme, setTheme] = React.useState({name: "", description: ""});

  useEffect(() => {
    client.get('/problems/coding-questions?theme=' + idPage.id)
      .then(res => {
        setQueryData(res.data);
      });
      client.get(`/problems/themes/` + idPage.id)
        .then(res => {
          setTheme(res.data);
        });
  }, [idPage.id]);

  return (
    <Grid
      container
      direction="row"
      justifyContent="center"
      alignItems="center"
      marginTop={'8vh'}>
      <Grid>
      </Grid>
          <Grid item xs={8} alignItems="center">
            <TableContainer component={Paper}>
              <Table aria-label="collapsible table">
                <TableHead>
                  <TableRow>
                    <TableCell/>
                    <TableCell>
                      <Typography variant="h6" gutterBottom component="div" textAlign="center">
                        {theme.name}
                      </Typography>
                      <Typography gutterBottom component="div" textAlign="justify">
                        {theme.description}
                      </Typography>
                    </TableCell>
                    <TableCell/>
                  </TableRow>
                </TableHead>
                <TableBody>
                  {queryData.map((row, index) => {                  
                    row.backgroundColor = (index%2 === 0 ? "#f5f5f5" : "white");
                    return <Row key={row.name} row={row} />
                  })}
                </TableBody>
              </Table>
            </TableContainer>
          </Grid>
        </Grid>
  );
}