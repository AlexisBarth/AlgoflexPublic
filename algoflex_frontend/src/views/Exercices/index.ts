import Exercices from "./components/Exercices";

export default Exercices;