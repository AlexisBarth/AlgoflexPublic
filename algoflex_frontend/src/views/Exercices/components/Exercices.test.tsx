import React from 'react';
import Exercices from './Exercices';
import { render, fireEvent } from "@testing-library/react";
import { act } from "react-dom/test-utils";

test.todo("Make Exercices tests");


jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
    useParams: () => ({
      pageId: "1234"
    }),
    useRouteMatch: () => ({ url: 'http://localhost:3000/theme/1234' }),
  }));


it("Renders correctly", () => {
    const { queryByTest } = render(<Exercices />);
    expect(queryByTest).toMatchSnapshot();
});

describe('Exercices', () => {
    it('Should work !', () => {
        render(<Exercices />);
    });
});