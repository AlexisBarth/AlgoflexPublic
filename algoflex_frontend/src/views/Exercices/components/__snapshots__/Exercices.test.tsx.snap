// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container css-1jvme59-MuiGrid-root"
      >
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vl5ofk-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
          >
            <table
              aria-label="collapsible table"
              class="MuiTable-root css-rqglhn-MuiTable-root"
            >
              <thead
                class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  />
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    <div
                      class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-baxsg6-MuiTypography-root"
                    />
                    <div
                      class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d6l48a-MuiTypography-root"
                    />
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  />
                </tr>
              </thead>
              <th
                class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                role="rowgroup"
              />
            </table>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container css-1jvme59-MuiGrid-root"
    >
      <div
        class="MuiGrid-root css-vj1n65-MuiGrid-root"
      />
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vl5ofk-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
        >
          <table
            aria-label="collapsible table"
            class="MuiTable-root css-rqglhn-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                />
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  <div
                    class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-baxsg6-MuiTypography-root"
                  />
                  <div
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d6l48a-MuiTypography-root"
                  />
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                />
              </tr>
            </thead>
            <th
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              role="rowgroup"
            />
          </table>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
